--- xutf8/configure.in~	2004-06-17 01:31:43.000000000 +0000
+++ xutf8/configure.in	2004-06-17 01:32:53.000000000 +0000
@@ -69,9 +69,10 @@
 AC_CHECK_LIB(X11, Xutf8DrawString,  ,)
 
 HAVE_LIBC_ICONV=0
-AC_CHECK_LIB(c, iconv, \
-        HAVE_LIBC_ICONV=1 ,\
-        )
+AC_CHECK_LIB(c, iconv, HAVE_LIBC_ICONV=1 ,)
+	if test $HAVE_LIBC_ICONV = 1; then
+	    LDFLAGS="$LDFLAGS -liconv"
+	fi
 AC_SUBST(HAVE_LIBC_ICONV)
 
 AC_OUTPUT(Makefile Xutf8.h)
