--- qiv/Makefile~	2004-05-22 07:21:47.000000000 +0000
+++ qiv/Makefile	2004-12-25 23:53:04.000000000 +0000
@@ -4,7 +4,7 @@
 #######################################################################
 
 # Directory where qiv will be installed under.
-PREFIX = /usr/local
+PREFIX := /usr/local
 
 # Font to use for statusbar in fullscreen mode
 STATUSBAR_FONT = "fixed"
@@ -57,13 +57,13 @@
 # Do not edit below here!
 ######################################################################
 
-CC        = gcc
-CFLAGS    = -O2 -Wall -fomit-frame-pointer -finline-functions \
+CC        := gcc
+CFLAGS    := -O2 -Wall -fomit-frame-pointer -finline-functions \
 	    -fcaller-saves -ffast-math -fno-strength-reduce \
 	    -fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
 
 INCLUDES  = `imlib-config --cflags-gdk`
-LIBS      = `imlib-config --libs-gdk`
+LIBS      = `imlib-config --libs-gdk` $(OPT_LIBS)
 
 PROGRAM   = qiv
 OBJS      = main.o image.o event.o options.o utils.o
@@ -82,7 +82,7 @@
 endif
 
 ifdef GTD_XINERAMA
-LIBS	 += -L/usr/X11R6/lib -lXinerama
+LIBS	 += -lXinerama
 endif
 
 PROGRAM_G = qiv-g
@@ -124,7 +124,7 @@
 
 install: $(PROGRAM)
 	@echo "Installing QIV..."
-	install -s -m 0755 $(PROGRAM) $(PREFIX)/bin
+	install -m 0755 $(PROGRAM) $(PREFIX)/bin
 	install -m 0644 $(PROGRAM).1 $(PREFIX)/man/man1
 	$(COMPRESS_PROG) $(PREFIX)/man/man1/$(PROGRAM).1
 	@if ./qiv -f ./intro.jpg ; \
