--- matchbox-desktop-0.8/src/mbdesktop.c~	2004-02-06 23:41:25.000000000 +0000
+++ matchbox-desktop-0.8/src/mbdesktop.c	2004-07-12 22:40:48.000000000 +0000
@@ -82,6 +82,15 @@
 
 #endif
 
+static void 
+sig_hup_reload_handler(int sig, siginfo_t *si, void *data)
+{
+  DBG("SIGHUP: Really Reload menu callback\n");
+
+  WantReload = True;
+
+}
+
 #ifdef USE_XSETTINGS
 
 #define XSET_UNKNOWN    0
@@ -802,9 +811,7 @@
   MBDesktop *mb;
   XGCValues gv;
 
-#ifdef USE_DNOTIFY
   struct sigaction act;
-#endif
 
   char *font_def       = FONT_DESC;
   char *font_title_def = FONT_TITLE_DESC;
@@ -1036,8 +1043,9 @@
   
   mb->win_top_level = XCreateWindow(mb->dpy, mb->root, 
 		                    0, 0,  
-				    mb->desktop_width, 
-				    mb->desktop_height, 0, 
+				    mb->desktop_width , 
+				    mb->desktop_height , 
+				    0, 
 				    CopyFromParent,  
 				    CopyFromParent, 
 				    mb->pixbuf->vis,
@@ -1073,6 +1081,11 @@
   sigaction(SIGRTMIN, &act, NULL);
 #endif 
 
+  act.sa_sigaction = sig_hup_reload_handler;
+  sigemptyset(&act.sa_mask);
+  act.sa_flags = SA_SIGINFO;
+  sigaction(SIGHUP, &act, NULL);
+
   mb->type_register_cnt = ITEM_TYPE_CNT;
 
   /* get the window ready */
@@ -1081,7 +1094,7 @@
 	       KeyPress | KeyRelease | StructureNotifyMask |
 	       FocusChangeMask );
 
-  XSelectInput(mb->dpy, mb->root, PropertyChangeMask );
+  XSelectInput(mb->dpy, mb->root, PropertyChangeMask|StructureNotifyMask );
 
   XMapWindow(mb->dpy, mb->win_top_level);
 
