--- gnumeric/Makefile.in~	2004-12-21 02:20:19.000000000 +0000
+++ gnumeric/Makefile.in	2004-12-21 02:20:35.000000000 +0000
@@ -280,7 +280,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = . po src plugins icons templates doc schemas
+SUBDIRS = . po src plugins icons templates schemas
 
 # automake does not like subdirs that are not directly below
 #if WITH_WIN32
