diff -Nur bzflag-2.0.0.20050117/src/platform/XWindow.cxx bzflag-2.0.0.20050117-new/src/platform/XWindow.cxx
--- bzflag-2.0.0.20050117/src/platform/XWindow.cxx	2005-01-02 09:43:36.000000000 +0000
+++ bzflag-2.0.0.20050117-new/src/platform/XWindow.cxx	2005-01-24 14:45:48.000000000 +0000
@@ -13,12 +13,14 @@
 #include "XWindow.h"
 #include "XVisual.h"
 #include "OpenGLGState.h"
+/*
 #if defined(XF86VIDMODE_EXT)
 #  define USE_XF86VIDMODE_EXT
 #  define private c_private
 #  include <X11/extensions/xf86vmode.h>
 #  undef private
 #endif
+*/
 #include <assert.h>
 #include <math.h>
 #include <ctype.h>
