diff -Nur gpaint-0.2/src/Makefile.am gpaint-0.2.new/src/Makefile.am
--- gpaint-0.2/src/Makefile.am	2000-11-20 21:08:28.000000000 +0000
+++ gpaint-0.2.new/src/Makefile.am	2004-10-07 17:13:34.000000000 +0000
@@ -13,6 +13,6 @@
 	gtkscrollframe.c gtkscrollframe.h fileio.c fileio.h image_processing.c image_processing.h\
 	clipboard.c clipboard.h print.c print.h
 
-gpaint_CFLAGS += $(EXTRA_GNOME_CFLAGS)
+gpaint_CFLAGS = $(EXTRA_GNOME_CFLAGS)
 gpaint_LDADD = $(GNOME_LIBDIR) $(GNOMEUI_LIBS) $(INTLLIBS) $(EXTRA_GNOME_LIBS)
 
