--- gpe/base/gpe-calculator/Makefile~	2005-02-07 09:02:11.000000000 +0000
+++ gpe/base/gpe-calculator/Makefile	2005-02-07 09:02:32.000000000 +0000
@@ -1,5 +1,5 @@
 PACKAGE = gpe-calculator
-PREFIX = /usr/local
+PREFIX = /usr
 DEBUG = no
 CVSBUILD = yes
 VERSION = 0.2
@@ -43,7 +43,7 @@
 	install -D -m 644 $(PACKAGE).png $(DESTDIR)$(PREFIX)/share/pixmaps/$(PACKAGE).png
 	mkdir -p $(DESTDIR)$(PREFIX)/share/applications
 	install -m 644 $(PACKAGE).desktop $(DESTDIR)$(PREFIX)/share/applications
-	strip $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
+	$(STRIP) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
 
 clean:
 	rm -f $(PACKAGE) $(OBJS) $(DEPS)
