--- gpe/base/gpe-conf/Makefile~	2005-02-07 09:17:37.000000000 +0000
+++ gpe/base/gpe-conf/Makefile	2005-02-07 10:01:51.000000000 +0000
@@ -3,10 +3,9 @@
 CVSBUILD = yes
 DEBUG = no
 INSTALL = install
-PREFIX = /usr/local
+PREFIX = /usr
 DESTDIR = /
 PACKAGETOOL = no
-STRIP = strip
 MACHINE = unknown
 
 CFLAGS += -MD
@@ -47,8 +46,8 @@
 	gpe-conf-sound.desktop.in
 
 ifeq ($(CVSBUILD),yes)
-PACKAGE_CPPFLAGS += -I../libxsettings -I../libxsettings-client
-PACKAGE_LDFLAGS += -L../libxsettings -L../libxsettings-client
+PACKAGE_CFLAGS += -I../libgpewidget
+PACKAGE_LDFLAGS += -L../libgpewidget
 BUILD = ../build
 else
 BUILD = build
@@ -59,7 +58,8 @@
 
 PACKAGE_CPPFLAGS += $(STANDARD_CPPFLAGS) -I. 
 PACKAGE_CFLAGS += $(STANDARD_CFLAGS) $(GPECFLAGS) $(GTKCFLAGS) -I.
-PACKAGE_LDFLAGS += $(STANDARD_LDFLAGS) $(GPELIBS) $(GTKLIBS)-lXsettings -lXsettings-client -lcrypt -lXrandr 
+PACKAGE_LDFLAGS += $(STANDARD_LDFLAGS) $(GPELIBS) $(GTKLIBS) -lXsettings-client -lcrypt -lXrandr -lgpewidget
+#  -lXsettings 
 
 ifeq ($(DEBUG),yes)
 PACKAGE_CFLAGS += -g -DDEBUG
