--- gpe/base/gpe-word/Makefile~	2005-02-07 02:43:44.000000000 +0000
+++ gpe/base/gpe-word/Makefile	2005-02-07 02:47:11.000000000 +0000
@@ -1,4 +1,4 @@
-PREFIX = /usr/local
+PREFIX = /usr
 PACKAGE = gpe-word
 VERSION=0.2
 DEBUG = no
@@ -28,7 +28,7 @@
 
 LDFLAGS += `pkg-config --libs gtk+-2.0`
 
-CFLAGS += -I../libdisplaymigration
+CFLAGS += -I../libdisplaymigration/libdisplaymigration
 LDFLAGS += -L../libdisplaymigration -ldisplaymigration
 
 CFLAGS += -DPACKAGE=\"$(PACKAGE)\" -DPREFIX=\"$(PREFIX)\" -DPACKAGE_LOCALE_DIR=\"$(PREFIX)/share/locale\"
@@ -46,7 +46,7 @@
 	install -D -m 644 $(PACKAGE).png $(DESTDIR)$(PREFIX)/share/pixmaps/$(PACKAGE).png
 	mkdir -p $(DESTDIR)$(PREFIX)/share/applications
 	install -m 644 $(PACKAGE).desktop $(DESTDIR)$(PREFIX)/share/applications/$(PACKAGE).desktop
-	strip $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
+	$(STRIP) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
 
 clean:
 	rm -f $(PACKAGE) $(OBJS)
