--- perl/ext/Errno/Errno_pm.PL~	2004-10-14 16:08:42.000000000 +0000
+++ perl/ext/Errno/Errno_pm.PL	2004-10-14 16:09:19.000000000 +0000
@@ -109,8 +109,8 @@
 	     ) {
 	# Some Linuxes have weird errno.hs which generate
 	# no #file or #line directives
-	my $linux_errno_h = -e '/usr/include/errno.h' ?
-	    '/usr/include/errno.h' : '/usr/local/include/errno.h';
+	my $linux_errno_h = -e '@CROSS_LIB_DIR@/include/errno.h' ?
+	    '@CROSS_LIB_DIR@/include/errno.h' : '/usr/local/include/errno.h';
 	$file{$linux_errno_h} = 1;
     } elsif ($^O eq 'MacOS') {
 	# note that we are only getting the GUSI errno's here ...
