--- pkgconfig/glib-1.2.8/configure~	2003-01-16 21:09:16.000000000 +0000
+++ pkgconfig/glib-1.2.8/configure	2004-05-10 18:42:18.000000000 +0000
@@ -2079,7 +2079,7 @@
 glib_save_LIBS=$LIBS
 LIBS="$LIBS -lm"
 if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+    { echo "configure: error: can not run test program while cross compiling" 1>&2; }
 else
   cat > conftest.$ac_ext <<EOF
 #line 2086 "configure"
@@ -2524,7 +2524,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+    ac_cv_sizeof_char=1
 else
   cat > conftest.$ac_ext <<EOF
 #line 2531 "configure"
@@ -2563,7 +2563,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  ac_cv_sizeof_short=2
 else
   cat > conftest.$ac_ext <<EOF
 #line 2570 "configure"
@@ -2602,7 +2602,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  ac_cv_sizeof_long=4
 else
   cat > conftest.$ac_ext <<EOF
 #line 2609 "configure"
@@ -2641,7 +2641,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  ac_cv_sizeof_int=4
 else
   cat > conftest.$ac_ext <<EOF
 #line 2648 "configure"
@@ -2680,7 +2680,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  ac_cv_sizeof_void_p=4
 else
   cat > conftest.$ac_ext <<EOF
 #line 2687 "configure"
@@ -2834,19 +2834,19 @@
 if eval "test \"`echo '$''{'glib_cv_has__inline'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
-  
-        if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
-else
+
   cat > conftest.$ac_ext <<EOF
 #line 2843 "configure"
 #include "confdefs.h"
 
 	__inline int foo () { return 0; }
 	int main () { return foo (); }
+
+
+
        	
 EOF
-if { (eval echo configure:2850: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
+if { (eval echo configure:2850: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }
 then
   glib_cv_has__inline=yes
         
@@ -2858,8 +2858,6 @@
         
 fi
 rm -fr conftest*
-fi
-
 
 fi
 
@@ -2875,19 +2873,19 @@
 if eval "test \"`echo '$''{'glib_cv_has__inline__'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
-  
-        if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
-else
+
   cat > conftest.$ac_ext <<EOF
 #line 2884 "configure"
 #include "confdefs.h"
 
 	__inline__ int foo () { return 0; }
 	int main () { return foo (); }
+
+
+
        	
 EOF
-if { (eval echo configure:2891: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
+if { (eval echo configure:2891: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }
 then
   glib_cv_has__inline__=yes
         
@@ -2899,7 +2897,6 @@
         
 fi
 rm -fr conftest*
-fi
 
 
 fi
@@ -2916,16 +2913,16 @@
 if eval "test \"`echo '$''{'glib_cv_hasinline'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
-  
-        if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
-else
+
   cat > conftest.$ac_ext <<EOF
 #line 2925 "configure"
 #include "confdefs.h"
 
 	inline int foo () { return 0; }
 	int main () { return foo (); }
+
+
+
        	
 EOF
 if { (eval echo configure:2932: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
@@ -2940,7 +2937,6 @@
         
 fi
 rm -fr conftest*
-fi
 
 
 fi
@@ -3895,7 +3891,11 @@
 else
   
 	if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to yes" 1>&2
+	 glib_cv_sane_realloc=yes
+	 cat >> confdefs.h <<\EOF
+#define REALLOC_0_WORKS 1
+EOF
 else
   cat > conftest.$ac_ext <<EOF
 #line 3902 "configure"
@@ -3937,7 +3937,9 @@
 else
   
 	if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to no" 1>&2
+	glib_cv_va_copy=no
+	
 else
   cat > conftest.$ac_ext <<EOF
 #line 3944 "configure"
@@ -3982,7 +3984,8 @@
 else
   
 	if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to yes" 1>&2
+	glib_cv___va_copy=yes
 else
   cat > conftest.$ac_ext <<EOF
 #line 3989 "configure"
@@ -4027,7 +4030,8 @@
 else
   
 	if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to yes" 1>&2
+	glib_cv_va_val_copy=yes
 else
   cat > conftest.$ac_ext <<EOF
 #line 4034 "configure"
@@ -4381,7 +4385,8 @@
 else
   
 		if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to no" 1>&2
+			glib_cv_uscore=no
 else
   cat > conftest.$ac_ext <<EOF
 #line 4388 "configure"
@@ -5211,7 +5216,8 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to yes" 1>&2
+				ac_cv_func_getpwuid_r=yes
 else
   cat > conftest.$ac_ext <<EOF
 #line 5218 "configure"
@@ -5311,7 +5317,10 @@
 echo "configure:5312: checking whether pthread_mutex_trylock is posix like" >&5
 		# DCE Threads return 1 as success, posix 0. what a mess.
 		if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to yes" 1>&2
+			cat >> confdefs.h <<\EOF
+#define HAVE_PTHREAD_MUTEX_TRYLOCK_POSIX 1
+EOF
 else
   cat > conftest.$ac_ext <<EOF
 #line 5318 "configure"
@@ -5341,7 +5350,10 @@
 echo "configure:5342: checking whether pthread_cond_timedwait is posix like" >&5
 		# DCE Threads return -1 as failure, posix ETIMEDOUT.
 		if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  echo "configure: warning: cannot check it if cross compiling -- defaulting to yes" 1>&2
+			cat >> confdefs.h <<\EOF
+#define HAVE_PTHREAD_COND_TIMEDWAIT_POSIX 1
+EOF
 else
   cat > conftest.$ac_ext <<EOF
 #line 5348 "configure"
@@ -5406,7 +5418,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  glib_cv_sizeof_gmutex=24
 else
   cat > conftest.$ac_ext <<EOF
 #line 5413 "configure"
@@ -5526,7 +5538,17 @@
 done
 echo "return 0; }" >>confrun.c
 if test "$cross_compiling" = yes; then
-    { echo "configure: error: can not run test program while cross compiling" 1>&2; exit 1; }
+  	for glib_sysdef_input in POLLIN:1 POLLOUT:4 POLLPRI:2 POLLERR:8 POLLHUP:16 POLLNVAL:32 ; do
+		glib_sysdef=`echo $glib_sysdef_input | sed 's/^\([^:]*\):.*$/\1/'`
+		glib_default=`echo $glib_sysdef_input | sed 's/^[^:]*:\(.*\)$/\1/'`
+		if test $glib_sysdef != $glib_default; then
+			glib_default=" =$glib_default"
+		else
+			glib_default=
+		fi
+		echo "#define GLIB_SYSDEF_$glib_sysdef$glib_default" >>$glib_sysdefso
+	done
+	echo "configure: warning: cannot check it if cross compiling -- using defaults" 1>&2
 else
   cat > conftest.$ac_ext <<EOF
 #line 5533 "configure"
