--- src/configure~	2004-09-25 21:14:09.000000000 +0000
+++ src/configure	2004-09-25 18:21:22.000000000 +0000
@@ -5081,7 +5081,7 @@
 
   if test "x$kde_use_qt_emb" != "xyes"; then
   __val=$CXX
-  __forbid=" -fno-rtti -rpath "
+  __forbid=" -rpath "
   if test -n "$__val"; then
     __new=""
     ac_save_IFS=$IFS
@@ -5100,7 +5100,7 @@
   fi
 
   __val=$CXXFLAGS
-  __forbid=" -fno-rtti -rpath "
+  __forbid=" -rpath "
   if test -n "$__val"; then
     __new=""
     ac_save_IFS=$IFS
