diff -Nur matchbox-panel-0.8/applets/mb-applet-menu-launcher.c matchbox-panel-0.8-patched/applets/mb-applet-menu-launcher.c
--- matchbox-panel-0.8/applets/mb-applet-menu-launcher.c	2004-02-07 17:48:01.000000000 +0000
+++ matchbox-panel-0.8-patched/applets/mb-applet-menu-launcher.c	2004-02-17 23:27:45.000000000 +0000
@@ -366,6 +367,15 @@
 }
 #endif
 
+static 
+void reload_hup_menu(int signum, siginfo_t *siginfo, void *data)
+{
+  DBG("HUP: Reload menu callback\n");
+
+  WantReload = True;
+
+}
+
 static void
 build_menu(void)
 {
@@ -968,6 +980,11 @@
    act.sa_handler = reap_children;
    sigaction(SIGCHLD, &act, NULL);
 
+   act.sa_sigaction = reload_hup_menu;
+   sigemptyset(&act.sa_mask);
+   act.sa_flags = SA_SIGINFO;
+   sigaction(SIGHUP, &act, NULL);
+
 #ifdef USE_DNOTIFY
 
 #define DD_DIR DATADIR "/applications"
