--- nxkbd/configure.ac~	2004-10-04 04:35:18.000000000 +0000
+++ nxkbd/configure.ac	2004-10-04 04:36:04.000000000 +0000
@@ -7,8 +7,8 @@
 AC_ARG_ENABLE(xft,
   [  --enable-xft           enable xft (anti-aliased) support],
   [if test x$enable_xft = xyes; then
-      CFLAGS="$CFLAGS -DUSE_XFT"
-      X_LIBS="$X_LIBS -lXft"
+      CFLAGS="$CFLAGS -DUSE_XFT `xft-config --cflags`"
+      X_LIBS="$X_LIBS `xft-config --libs`"
   fi])
 
 dnl check for xft support
