--- nxcomp/configure.in~	2004-10-03 21:18:11.000000000 +0000
+++ nxcomp/configure.in	2004-10-03 21:23:02.000000000 +0000
@@ -57,7 +57,7 @@
 AC_MSG_CHECKING([whether compiler needs -Wno-deprecated])
 gcc_version=`${CC} --version | grep 'gcc (GCC) 3.' | head -n 1`
 case "${gcc_version}" in
-     gcc*)
+     *gcc*)
         AC_MSG_RESULT([yes])
         CXXFLAGS="$CXXFLAGS -Wno-deprecated"
         CPPFLAGS="$CPPFLAGS -Wno-deprecated"
@@ -71,7 +71,7 @@
 AC_MSG_CHECKING([whether compiler accepts -Wmissing-declarations and -Wnested-externs])
 gcc_version=`${CC} --version | grep 'gcc (GCC) 3.3.' | head -n 1`
 case "${gcc_version}" in
-     gcc*)
+     *gcc*)
         AC_MSG_RESULT([no])
         ;;
 
