--- ROX-Session-0.1.22/ROX-Session/setup.py~	2004-11-16 00:08:54.000000000 +0000
+++ ROX-Session-0.1.22/ROX-Session/setup.py	2004-11-16 00:09:24.000000000 +0000
@@ -78,7 +78,7 @@
 # Load a window manager. Keep trying until we find one that works!
 
 for wm in xfwm4 sawfish sawmill enlightenment wmaker icewm blackbox fluxbox \\
-	metacity kwin kwm fvwm2 fvwm 4Dwm twm; do
+	metacity openbox kwin kwm fvwm2 fvwm 4Dwm twm; do
   if [ -x "`which $wm`" ]; then break; fi;
 done
 
