diff -Nur licq-1.3.0-PRE/bin/viewurl-ncftp.sh licq-1.3.0-PRE.new/bin/viewurl-ncftp.sh
--- licq-1.3.0-PRE/bin/viewurl-ncftp.sh	2000-07-06 01:47:18.000000000 +0000
+++ licq-1.3.0-PRE.new/bin/viewurl-ncftp.sh	1970-01-01 00:00:00.000000000 +0000
@@ -1,14 +0,0 @@
-#!/bin/sh
-#
-# A shell script to view a url using lynx and xterm
-# Set the XTERM variable to where your xterm binary is and 
-# the LYNX variable to where your lynx binary is or
-# leave them as is if the binaries are in your path.
-# 
-# Graham Roff, February 1999.
-#
-
-NCFTP=ncftp
-XTERM=xterm
-
-${XTERM} -title "Licq URL View" -e ${NCFTP} "$1"
