diff -Nur pango-1.2.4/modules/indic/indic-ot.h pango-1.2.4.new/modules/indic/indic-ot.h
--- pango-1.2.4/modules/indic/indic-ot.h	2003-08-08 17:05:01.000000000 +0000
+++ pango-1.2.4.new/modules/indic/indic-ot.h	2004-08-24 04:35:59.000000000 +0000
@@ -9,7 +9,8 @@
 #ifndef __INDIC_OT_H__
 #define __INDIC_OT_H__
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <pango/pango-glyph.h>
 #include <pango/pango-types.h>
 #include "mprefixups.h"
