--- pango/pango/opentype/pango-ot-info.c~	2004-08-24 04:43:15.000000000 +0000
+++ pango/pango/opentype/pango-ot-info.c	2004-08-24 04:43:43.000000000 +0000
@@ -21,8 +21,8 @@
 
 #include "pango-ot-private.h"
 #include "fterrcompat.h"
-#include <freetype/internal/ftobjs.h>
-#include <freetype/ftmodule.h>
+#include FT_INTERNAL_OBJECTS_H
+#include FT_MODULE_H
 
 static void pango_ot_info_class_init (GObjectClass *object_class);
 static void pango_ot_info_finalize   (GObject *object);
